src/native/fs/fs_napi.cpp

```diff
@@ -991,6 +995,28 @@ struct FileSetxattr : public FSWrapWorker<FileWrap>
     }
 };
 
+struct LinkFileAt : public FSWrapWorker<FileWrap>
+{
+    std::string _filepath;
+
+    LinkFileAt(const Napi::CallbackInfo& info)
+        : FSWrapWorker<FileWrap>(info)
+    {
+        _filepath = info[2].As<Napi::String>();
+        Begin(XSTR() << "LinkFileAt " << DVAL(_wrap->_path) << DVAL(_wrap->_fd) << DVAL(_filepath));
+    }
+    virtual void Work()
+    {
+        int _rc_res;
+        // Link the Linux linkat() if there is no old path use empty string.
```

Comment:
can you explain in that comment the difference between the gpfs_linkat version and the linux linkat version?
The man page explains the same mechanism for linkat - https://man7.org/linux/man-pages/man2/link.2.html

>AT_EMPTY_PATH (since Linux 2.6.39)
              If oldpath is an empty string, create a link to the file
              referenced by olddirfd (which may have been obtained using
              the [open(2)](https://man7.org/linux/man-pages/man2/open.2.html) O_PATH flag).  In this case, olddirfd can
              refer to any type of file except a directory.  This will
              generally not work if the file has a link count of zero
              (files created with O_TMPFILE and without O_EXCL are an
              exception).  The caller must have the CAP_DAC_READ_SEARCH
              capability in order to use this flag.  This flag is Linux-
              specific; define _GNU_SOURCE to obtain its definition.

Reply:
Linux stat the file name and if it doesn't exist (which it will not, we create temp file with no name) it will fail the link call before calling the underling fs (in this case gpfs) that is the only reason we added the call, the interface is identical.

Reply:
I understand what you say, but the man for linkat says that with AT_EMPTY_PATH option it will allow an empty oldpath and then treat olddirfd as file fd, which is what we need, no?

Reply:
That is not the problem, Linux stat the file name of the name to be linked, it is a problem with Linux that the link operation don't get the to underline fs but we don't have an option here.

Reply:
Oh right right right, so the entire thing is because we need to support replacing an existing name, and linux linkat will not allow it to pass, correct?

Reply:
yes, I think I got it backward, it check that the file doesn't exist, and if it does it fails, and we need to also replace existing files.

Reply:
got it. so lets just update the comment to explain that reason.

Reply:
done

Reply:
thanks!